src/cst/parse.js

```diff
@@ -30,5 +38,25 @@ export default function parse(src) {
     return true
   }
   documents.toString = () => documents.join('...\n')
+  if (options.computeLineOffsets) documents.lineOffsets = lf
   return documents
 }
+
+export function charPosToLineCol(offset, lineOffsets) {
+  if (
+    !lineOffsets ||
+    typeof offset === 'undefined' ||
+    offset < 0 ||
+    offset > lineOffsets[lineOffsets.length - 1]
+  )
+    return { line: undefined, col: undefined }
+  const lineIndex = lineOffsets.indexOf(offset)
+  if (lineIndex >= 0)
+    return { line: lineIndex, col: offset - lineOffsets[lineIndex] }
```

Comment:
This is special-case handling for when `offset` points to a `\n` character, yes? Why couldn't it be handled by the following `for` loop?

Reply:
I may have optimised pre-emptively here (which is a mistake without measuring it). It just avoids the search-in-Javascript and allows the optimised `indexOf` method to have a chance. As the array is always sorted, perhaps a binary search-like approach would be better than this plus the loop, especially for large documents?

Reply:
I'd recommend first going with just a simple for loop, and then seeing if there's a real-world case that would show a need for a more complicated search pattern.